Errors page (#28)
transfeera/api-doc: errors.md

# Errors

The errors listed below are grouped by context and may be printed in different situations
They contain:

> - Code - The code starting by the initial of the error context, example: DBA_1 which belongs to DBA context.

Reviewer: ```suggestion
- Code - The code starts with the initials of the error context, example: DBA_1 which belongs to DBA context.
```

- Message - The message of the error.
- Field - The field that the error may be linked to.
- Variable - A variable name used to compile (by our API) a error message to show additional info.
- Data - The data used to compile the error message, example: { maxLength: 10 }. In this example a error message that contain "{{maxLength}}", will have this text replaced with the value on the data field, for this, a variable name with the same name of the key on the data object must be declared.

## Contexts

- [Destination Bank Account](#destination-bank-account)

> - [Transfers](#tranfers)

Reviewer: ```suggestion
- [Transfers](#transfers)
```

| DBA_7  | Account is required and must have between {{minLength}} and {{maxLength}} numbers | account       | ['maxLength', 'minLength'] |
| DBA_8  | Account digit is required and must be only 1 digit                                | account_digit |                            |
| DBA_9  | Account is required and must contain exactly {{exactLength}} numbers              | account       | exactLength                |
| DBA_10 | Invalid account type                                                              | account_type  |                            |
| DBA_11 | Account type 'Entidade pública' is only available for CNPJ                        | account_type  |                            |
| DBA_12 | Destination bank account document is not a CPF nor CNPJ                           | cpf_cnpj      |                            |
| DBA_13 | Destination bank account document is blocked                                      | cpf_cnpj      |                            |
| DBA_14 | Bank(s) not found by id                                                           | bank          |                            |

> ## Tranfers

Reviewer: ```suggestion
## Transfers
```

# Errors

The errors listed below are grouped by context and may be printed in different situations
They contain:
- Code - The code starts with the initials of the error context, example: DBA_1 which belongs to DBA context.
- Message - The message of the error.

> - Field - The field that the error may be linked to.

Reviewer 2: `additional info`

> - Variable - A variable name used to compile (by our API) a error message to show additional info.

Reviewer 2: `additional info`

> - Data - The data used to compile the error message, example: { maxLength: 10 }. In this example a error message that contain "{{maxLength}}", will have this text replaced with the value on the data field, for this, a variable name with the same name of the key on the data object must be declared.

Reviewer 2: `additional info` something like that.

#### Example of response
```
{
  data: {
    maxLength: 8,
    minLength: 7
  },
  error: "Bad Request",
  errorCode: "DBA_6",
  field: "account",
  message: "Account is required and must have between 7 and 8 numbers (Agency: 0001 / Account: 111111 / Value: 10)",
  statusCode: 400
}
```

## Contexts

- [Destination Bank Account](#destination-bank-account)
- [Transfers](#transfers)
- [Batch](#batch)
- [System](#system)

## Destination Bank Account

> | Code     | message                                                                           | field         | variable                   |

Reviewer 2: only `code` and `message` is enough.

|----------|-----------------------------------------------------------------------------------|---------------|----------------------------|
| DBA_1    | Agency digit must be numeric or x                                                 | agency_digit  |                            |
| DBA_2    | Account digit is required and must be numeric or x                                | account_digit |                            |

> | DBA_3    | Agency must be 1                                                                  | account       |                            |

Reviewer 2: field `account` ?

| DBA_8    | Account digit is required and must be only 1 digit                                | account_digit |                            |
| DBA_9    | Account is required and must contain exactly {{exactLength}} numbers              | account       | exactLength                |
| DBA_10   | Invalid account type                                                              | account_type  |                            |
| DBA_11   | Account type 'Entidade pública' is only available for CNPJ                        | account_type  |                            |
| DBA_12   | Destination bank account document is not a CPF nor CNPJ                           | cpf_cnpj      |                            |
| DBA_13   | Destination bank account document is blocked                                      | cpf_cnpj      |                            |
| DBA_14 * | Bank(s) not found by id                                                           | bank          |                            |

- `*` `bank_id` need to be filled with id of the bank returned from `/bank` endpoint

## Transfers

> | Code    | message                                                                   | field    | variable     |

Reviewer 2: only `code` and `message` is enough.

|---------|---------------------------------------------------------------------------|----------|--------------|
| TRA_1   | Transfer value can not be R$0,00                                          | value    |              |
| TRA_2 * | Transfer value for this bank can not be greater than R${{bankMaxValue}}   | value    | bankMaxValue |
| TRA_3   | Wrong format of a transfer field content                                  | transfer |              |

- `*` Default limit is at maximum R$ 25.000,00, if you need more than that please contact us

## Batch

> | Code  | message                                                                   | field        | variable     |

Reviewer 2: only `code` and `message` is enough.

|-------|---------------------------------------------------------------------------|--------------|--------------|
| BAT_1 | Your batch has no transfers, register transfers to close it.              | transfers    |              |

## System

> | Code  | message                                                                   | field        | variable     |

Reviewer 2: only `code` and `message` is enough.

|-------|---------------------------------------------------------------------------|--------------|--------------|
| SYS_1 * | An error has occurred                                                     | message      |              |

> - `*` If this error persits, please contact us

Reviewer: ```suggestion
- `*` If this error persists, please contact us
```

> - `*` If this error persists, please contact us

Reviewer 2: `If this error occur, please contact us`
